p2p/host/resource-manager/conn_limiter.go

```diff
@@ -16,81 +29,177 @@ type ConnLimitPerCIDR struct {
 // down
 var defaultMaxConcurrentConns = 8
 
-var defaultIP4Limit = ConnLimitPerCIDR{
-	ConnCount: defaultMaxConcurrentConns,
-	BitMask:   32,
+var defaultIP4Limit = ConnLimitPerSubnet{
+	ConnCount:    defaultMaxConcurrentConns,
+	PrefixLength: 32,
 }
-var defaultIP6Limits = []ConnLimitPerCIDR{
+var defaultIP6Limits = []ConnLimitPerSubnet{
 	{
-		ConnCount: defaultMaxConcurrentConns,
-		BitMask:   56,
+		ConnCount:    defaultMaxConcurrentConns,
+		PrefixLength: 56,
 	},
 	{
-		ConnCount: 8 * defaultMaxConcurrentConns,
-		BitMask:   48,
+		ConnCount:    8 * defaultMaxConcurrentConns,
+		PrefixLength: 48,
 	},
 }
 
-func WithLimitPeersPerCIDR(ipv4 []ConnLimitPerCIDR, ipv6 []ConnLimitPerCIDR) Option {
```

Comment:
We can't change this name for the patch release. I'm fine with just making a new release. v0.35.

Reply:
On the same topic can we introduce a new Option in a patch release(`WithNetworkPrefixLimit`)? It's backwards compatible but definitely not a bug fix.

Reply:
Okay, let's make a new release to v0.35.

> On the same topic can we introduce a new Option in a patch release(WithNetworkPrefixLimit)? It's backwards compatible but definitely not a bug fix.

Technically you're right. It's not a patch release per semver. We don't have a great way of distinguishing between major and minor releases right now though and I'd argue a minor release is closer to a patch release in spirit than a minor release is to a major relase.